src/gretel_trainer/benchmark/comparison.py

```diff
@@ -318,15 +354,18 @@ def _make_dataset(dataset: DatasetTypes, working_dir: Path) -> Dataset:
         row_count=dataset.row_count,
         column_count=dataset.column_count,
         data_source=source,
+        public=dataset.public,
     )
 
 
-def _make_run_identifier(
-    model: Union[GretelModel, CustomModel], dataset: Dataset
-) -> str:
-    run_identifier = f"{_model_name(model)}{RUN_IDENTIFIER_SEPARATOR}{dataset.name}"
-    log(run_identifier, "preparing run")
-    return run_identifier
+def _make_run_key(model: Union[GretelModel, CustomModel], dataset: Dataset) -> Tuple[str, str]:
+    run_key = (_model_name(model), dataset.name)
+    return run_key
+
+
+def _make_run_identifier(run_key: Tuple[str, str]) -> str:
+    model_name, dataset_name = run_key
+    return RUN_IDENTIFIER_SEPARATOR.join((model_name, dataset_name))
```

Comment:
If we (re-)make a dedicated `RunIdentifier` class, this could be a property of the object (currently returned by `_make_run_key`)